opentelemetry-sdk/src/opentelemetry/sdk/trace/_sampling_experimental/_traceid_ratio.py

```diff
@@ -0,0 +1,80 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Sequence
+
+from opentelemetry.context import Context
+from opentelemetry.trace import Link, SpanKind, TraceState
+from opentelemetry.util.types import Attributes
+
+from ._composable import ComposableSampler, SamplingIntent
+from ._trace_state import serialize_th
+from ._util import INVALID_THRESHOLD, MAX_THRESHOLD, calculate_threshold
+
+
+class ComposableTraceIDRatioBased(ComposableSampler):
+    _threshold: int
+    _description: str
+
+    def __init__(self, ratio: float):
+        threshold = calculate_threshold(ratio)
+        if threshold == MAX_THRESHOLD:
+            threshold_str = "max"
+        else:
+            threshold_str = serialize_th(threshold)
+        if threshold != MAX_THRESHOLD:
+            intent = SamplingIntent(threshold=threshold)
+        else:
+            intent = SamplingIntent(
+                threshold=INVALID_THRESHOLD, threshold_reliable=False
+            )
+        self._intent = intent
+        self._description = f"ComposableTraceIDRatioBased{{threshold={threshold_str}, ratio={ratio}}}"
+
+    def sampling_intent(
+        self,
+        parent_ctx: Context | None,
+        name: str,
+        span_kind: SpanKind | None,
+        attributes: Attributes,
+        links: Sequence[Link] | None,
+        trace_state: TraceState | None,
+    ) -> SamplingIntent:
+        return self._intent
+
+    def get_description(self) -> str:
+        return self._description
+
+
+def composable_traceid_ratio_based(
```

Comment:
Do you think it's premature to entry points for them already in pyproject so that they can be configured from env vars? When we do so we should also teach `_import_sampler` from `src/opentelemetry/sdk/_configuration/__init__.py` about them and `_KNOWN_SAMPLERS` / `_get_from_env_or_default` from the current sampling module.

Also is there a reason you are exposing this functions instead of the classes? The current samplers just expose the class.

Reply:
I think since these are experimental we can wait and just keep them manual for now.